tests/test_ec2.py

```diff
@@ -904,6 +904,25 @@ def test_ec2_add_security_groups(self):
         self.assertEqual(len(
             second_resources[0]['NetworkInterfaces'][0]['Groups']), 2)
 
+class TestAutoRecoverAlarmAction(BaseTest):
+    def test_autorecover_alarm(self):
+        session_factory = self.replay_flight_data('test_ec2_autorecover_alarm')
+        p = self.load_policy(
+            {'name': 'ec2-autorecover-alarm',
+             'resource': 'ec2',
+             'filters': [
+                 {'tag:Name': 'Singleton'},
+                 {'type': 'singleton'}],
+             'actions': [
+                 {'type': 'autorecover-alarm'}]},
+            config={'region': 'us-west-1'},
+            session_factory=session_factory)
+
+        resources = p.run()
+
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]['InstanceId'], 'i-00fe7967fb7167c62')
```

Comment:
this doesn't actually assert/verify the effect of the action which would be creating the named alarm "recover-{}'.format(i['InstanceId'])",  

as pseudo example of verifying
```
client = session_factory('cloudwatch')
try:
    result = client.describe_alarms(
         AlarmNames=['recover-{}'.format(resources[0]['InstanceId']])
    self.assertTrue(result.get('MetricAlarms')) 
except:
    self.fail("alarm not found")
```